tapchannel/aux_leaf_signer_test.go

```diff
@@ -163,3 +179,287 @@ func makeCommitSig(t *testing.T, numAssetIDs, numHTLCs int) *lnwire.CommitSig {
+			name:   "index math.MaxUint64-2",
+			index:  math.MaxUint64 - 2,
+			result: maxUint64MinusOne,
+		},
+		{
+			name:   "index math.MaxUint64-1",
+			index:  math.MaxUint64 - 1,
+			result: maxUint64,
+		},
+		{
+			name:   "index math.MaxUint64, wraps around to 1",
+			index:  math.MaxUint64,
+			result: new(secp256k1.ModNScalar).SetInt(1),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tweak := HtlcIndexAsScriptKeyTweak(tc.index)
+			require.Equal(t, tc.result, tweak)
+		})
+	}
+}
+
+// TestTweakPubKeyWithIndex tests the TweakPubKeyWithIndex function.
+func TestTweakPubKeyWithIndex(t *testing.T) {
+	// We want a random number in the range of uint32 but will need it as
+	// an uint64 for the test cases.
+	randNum := uint64(test.RandInt[uint32]())
+	startKey := pubKeyFromUint64(randNum)
+
+	testCases := []struct {
+		name   string
+		pubKey *btcec.PublicKey
+		index  uint64
+		result *btcec.PublicKey
+	}{
+		{
+			name:   "index 0",
+			pubKey: startKey,
+			index:  0,
+			result: pubKeyFromUint64(randNum + 1),
+		},
+		{
+			name:   "index 1",
+			pubKey: startKey,
+			index:  1,
+			result: pubKeyFromUint64(randNum + 2),
+		},
+		{
+			name:   "index 99",
+			pubKey: startKey,
+			index:  99,
+			result: pubKeyFromUint64(randNum + 100),
+		},
+		{
+			name:   "index math.MaxUint32-1",
+			pubKey: startKey,
+			index:  math.MaxUint32 - 1,
+			result: pubKeyFromUint64(randNum + math.MaxUint32),
+		},
+		{
+			// Because we always increment by 1, there is a
+			// "collision" at 0 and math.MaxUint64. For the purpose
+			// of tweaking channel related keys with the HTLC index,
+			// that is okay, as there isn't expected to ever be that
+			// many HTLCs during the lifetime of a channel.
+			name:   "index math.MaxUint64, wrap around",
+			pubKey: startKey,
+			index:  math.MaxUint64,
+			result: pubKeyFromUint64(randNum + 1),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tweakedKey := TweakPubKeyWithIndex(tc.pubKey, tc.index)
+			require.Equal(
+				t, tc.result.SerializeCompressed(),
+				tweakedKey.SerializeCompressed(),
+			)
+		})
+	}
+}
+
+// TestTweakHtlcTree tests the TweakHtlcTree function.
+func TestTweakHtlcTree(t *testing.T) {
+	randTree := txscript.AssembleTaprootScriptTree(
+		test.RandTapLeaf(nil), test.RandTapLeaf(nil),
+		test.RandTapLeaf(nil),
+	)
+	randRoot := randTree.RootNode.TapHash()
+
+	// We want a random number in the range of uint32 but will need it as
+	// an uint64 for the test cases.
+	randNum := uint64(test.RandInt[uint32]())
+
+	makeTaprootKey := func(num uint64) *btcec.PublicKey {
+		return txscript.ComputeTaprootOutputKey(
+			pubKeyFromUint64(num), randRoot[:],
+		)
+	}
+	startKey := pubKeyFromUint64(randNum)
+	startTaprootKey := makeTaprootKey(randNum)
+
+	testCases := []struct {
+		name   string
+		tree   input.ScriptTree
+		index  uint64
+		result input.ScriptTree
+	}{
+		{
+			name: "index 0",
+			tree: input.ScriptTree{
```

Comment:
super-nit: `tree` is already the same here and handled as read-only, so could be deduplicated.